dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/LibertyValidatorFactoryProxy.java

```diff
@@ -0,0 +1,164 @@
+        ModuleMetaData mmd = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getModuleMetaData();
+        ValidatorFactory vf = validatorFactoryCache.get(mmd);
+        if (vf == null) {
+            vf = BeanValidationAccessor.getValidatorFactory();
+            validatorFactoryCache.put(mmd, vf);
+        }
+        return vf;
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#close()
+     */
+    @Override
+    public void close() {
+        for (ValidatorFactory vf : validatorFactoryCache.values()) {
+            vf.close();
+        }
+        validatorFactoryCache.clear();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getClockProvider()
+     */
+    @Override
+    public ClockProvider getClockProvider() {
+        return delegate().getClockProvider();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getConstraintValidatorFactory()
+     */
+    @Override
+    public ConstraintValidatorFactory getConstraintValidatorFactory() {
+        return delegate().getConstraintValidatorFactory();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getMessageInterpolator()
+     */
+    @Override
+    public MessageInterpolator getMessageInterpolator() {
+        return delegate().getMessageInterpolator();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getParameterNameProvider()
+     */
+    @Override
+    public ParameterNameProvider getParameterNameProvider() {
+        return delegate().getParameterNameProvider();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getTraversableResolver()
+     */
+    @Override
+    public TraversableResolver getTraversableResolver() {
+        return delegate().getTraversableResolver();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#getValidator()
+     */
+    @Override
+    public Validator getValidator() {
+        return delegate().getValidator();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#unwrap(java.lang.Class)
+     */
+    @Override
+    public <T> T unwrap(Class<T> type) {
+        return delegate().unwrap(type);
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see javax.validation.ValidatorFactory#usingContext()
+     */
+    @Override
+    public ValidatorContext usingContext() {
+        return delegate().usingContext();
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see com.ibm.ws.container.service.metadata.ModuleMetaDataListener#moduleMetaDataCreated(com.ibm.ws.container.service.metadata.MetaDataEvent)
+     */
+    @Override
+    public void moduleMetaDataCreated(MetaDataEvent<ModuleMetaData> event) throws MetaDataException {
+        // no-op
+
+    }
+
+    /*
+     * (non-Javadoc)
```

Comment:
If you are making other changes to this file anyway, recommend removing "(non-Javadoc)" comment lines